src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,19 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= 35 && candidate.isAllowedToVote()
+                && checkDate(candidate) && candidate.getNationality().equals("Ukrainian");
+    }
+
+    private boolean checkDate(Candidate candidate) {
+        String[] arrayData = candidate.getPeriodsInUkr().split("-");
```

Comment:
```suggestion
                && checkDate(candidate.getPeriodsInUkr()) && candidate.getNationality().equals("Ukrainian");
    }

    private boolean checkDate(String periodInUkraine) {
        String[] arrayData = periodInUkraine.split(SEPARATOR);
```
pass only data that you process in this method